test/tap/tests/test_ps_hg_routing-t.cpp

```diff
@@ -60,6 +60,19 @@ int main(int argc, char** argv) {
 	MYSQL_QUERY(mysqladmin, "insert into mysql_query_rules (rule_id, active, flagIN, match_digest, negate_match_pattern, re_modifiers, destination_hostgroup, apply) values (200, 1, 0, '^SELECT', 0, 'CASELESS', 1, 1)");
 	MYSQL_QUERY(mysqladmin, "load mysql query rules to runtime");
 
+	for (;;) {
+		sleep(1);
+		ok(true, "Waiting for test.test1");
```

Comment:
Why are we using `ok()` ?
Is `ok()` used to track a successful test?

Reply:
I want to make logs of tests more verbose in case of problems. I cannot use fprintf, because of TAP protocol and yes, `ok` is not the best choice. After checking TAP API I found the `diag` function, which is better suited. This issue is fixed in the last commit.